src/utils/countSamples.ts

```diff
@@ -1,124 +1,110 @@
-        getNodeAttr(n, "location") !== homeGeo.location &&
-        nodeYear <= maxYear &&
-        nodeYear >= minYear &&
-        nodeMonth <= maxMonth &&
-        nodeMonth >= minMonth
+        // other states in country
+        location !== homeGeo.location &&
+        division !== homeGeo.division &&
+        country === homeGeo.country
       );
-    });
+    }
   }
-  return filteredNodes.length;
 };
 
-export const getGisaidCounts = (
-  gisaidRecords: GISAIDRecord[],
-  geoLevel: GeoLevels,
-  homeGeo: HomeGeo,
-  minYear: number,
-  minMonth: number,
-  maxYear: number,
-  maxMonth: number
+const matchesDateRange = (
+  type: "gisaid" | "node",
+  minDate: Date,
+  maxDate: Date,
+  gisaidRecord?: GISAIDRecord,
+  node?: Node
 ) => {
-  let filteredGisaidRecords: GISAIDRecord[] = [];
+  let date: Date;
 
-  if (geoLevel === "location") {
-    // same county
-    filteredGisaidRecords = gisaidRecords.filter(
-      (gr: GISAIDRecord) =>
-        gr.country === homeGeo.country &&
-        gr.division === homeGeo.division &&
-        gr.location === homeGeo.location &&
-        gr.year <= maxYear &&
-        gr.year >= minYear &&
-        gr.month - 1 <= maxMonth && // -1 for indexing (pandas dates are 1-idx; tsc dates are 0-idx)
-        gr.month - 1 >= minMonth
-    );
-  } else if (geoLevel === "division") {
-    // same state, different counties
-    filteredGisaidRecords = gisaidRecords.filter(
-      (gr: GISAIDRecord) =>
-        gr.country === homeGeo.country &&
-        gr.division === homeGeo.division &&
-        gr.location !== homeGeo.location &&
-        gr.year <= maxYear &&
-        gr.year >= minYear &&
-        gr.month - 1 <= maxMonth &&
-        gr.month - 1 >= minMonth
+  if (type === "gisaid" && gisaidRecord) {
+    date = new Date(Date.parse(`${gisaidRecord.year}-${gisaidRecord.month}`));
+  } else if (type === "node" && node) {
+    date = getNodeAttr(node, "num_date");
+  } else {
+    console.warn("ERROR: malformed input to matchesDateRange");
+    return false;
+  }
+
+  return date >= minDate && date <= maxDate;
+};
+
+export const getCount = (
+  type: "node" | "gisaid",
+  geoLevel: string,
+  homeGeo: HomeGeo,
+  minDate: Date,
+  maxDate: Date,
+  gisaidRecords?: GISAIDRecord[],
+  nodes?: Node[]
+) => {
+  if (type === "gisaid" && gisaidRecords) {
+    const filteredRecords = gisaidRecords.filter(
+      (record: GISAIDRecord) =>
+        matchesGeography("gisaid", homeGeo, geoLevel, record) &&
+        matchesDateRange("gisaid", minDate, maxDate, record)
     );
-  } else if (geoLevel === "country") {
-    // same country, different states
-    filteredGisaidRecords = gisaidRecords.filter(
-      (gr: GISAIDRecord) =>
-        gr.country === homeGeo.country &&
-        gr.division !== homeGeo.division &&
-        gr.location !== homeGeo.location &&
-        gr.year <= maxYear &&
-        gr.year >= minYear &&
-        gr.month - 1 <= maxMonth &&
-        gr.month - 1 >= minMonth
+
+    let total = 0;
```

Comment:
This is not actually necessary, but I wanted to point out this could be written as an accumulator / `reduce` instead of a for loop. It's basically summation on a single key, right? This works as is, but for some reason I was taught to be allergic to using for loops in JS. I'm not really sure why, but I always reach for filter/map/reduce before thinking about using a vanilla for loop when I'm writing JS. I'm not sure why I was taught that though, never really had to examine if that's actually true, so take that with a grain of salt.

Reply:
totally could. I just find `reduce` harder to think about / read.